test/e2e/spec/eid_oidc_provider/auth_spec.js

```diff
@@ -0,0 +1,108 @@
+describe("eId OpenId: The authorization endpoint", function() {
+    
+    var chakram = require('/usr/local/lib/node_modules/chakram/lib/chakram.js');
+    var expect = chakram.expect;
+    var URL = 'https://' + process.env.BOILERPLATE_DOMAIN + '/api/openid';
+
+    var response_type = process.env.OIDC_RESPONSE_TYPE;
+    var scope = process.env.OIDC_SCOPE;
+    var client_id = process.env.OIDC_CLIENT_ID;
+    var state = '32os93klas8923hf9qpw98qhf9f2q2q093fu';
+    var nonce = '2389fhq3034ht84f809hj348t943w94fcpm4';
+    var redirect_uri = process.env.OIDC_REDIRECT_URI;
+
+    it("does not redirect when client_id missing or invalid", function() {
+        let params = 'response_type=' + response_type
+            + '&scope=' + scope
+            + '&client_id=1'
+            + '&state=' + state
+            + '&nonce=' + nonce
+            + '&redirect_uri=' + encodeURIComponent(redirect_uri);
+        var request = URL + '/authorize/?' + params;
+        return chakram.get(request, {followRedirect: false})
+            .then(function(respObj) {
+                expect(respObj).to.have.status(200);
+                expect(respObj.response.body).to.match(/The client identifier \(client_id\) is missing or invalid/);
+                return chakram.wait();
+            });
+    });
+
+    it("does not redirect when redirect uri does not match to client_id", function() {
+        let params = 'response_type=' + response_type
+            + '&scope=' + scope
+            + '&client_id=' + client_id
+            + '&state=' + state
+            + '&nonce=' + nonce
+            + '&redirect_uri=' + encodeURIComponent("http://foo.de");
+        var request = URL + '/authorize?' + params;
+        return chakram.get(request, {followRedirect: false})
+            .then(function(respObj) {
+                expect(respObj).to.have.status(200);
+                expect(respObj.response.body).to.match(/Redirect URI Error/);
+                return chakram.wait();
+            });
+    });
+
+    it("redirects to redirect uri with error when scope is malformed", function() {
+        let params = 'response_type=' + response_type
+            + '&scope=foo'
+            + '&client_id=' + client_id
+            + '&state=' + state
+            + '&nonce=' + nonce
+            + '&redirect_uri=' + encodeURIComponent(redirect_uri);
+        var request = URL + '/authorize/?' + params;
+        var response = chakram.get(request, {followRedirect: false});
+        let regex = new RegExp(redirect_uri.replace(/(^\w+:|^)\/\//, '') + "#error=invalid_scope");
+        expect(response).to.have.status(302);
+        expect(response).to.have.header('Location', function(location) {
```

Comment:
You could just give [the regex as second argument](https://github.com/dareid/chakram/blob/master/lib/assertions/header.js#L31).